patroni/validator.py

```diff
@@ -369,14 +369,29 @@ def __init__(self, contains: OptionalType[List[str]] = None,
         self.contains = contains
         self.contains_executable = contains_executable
 
+    def _check_executables(self, path: OptionalType[str] = None) -> Union[str, None]:
+        """Check that all executables from contains_executable list exist within the given directory or within PATH.
+
+        :param path: optional path to the base directory against which executables will be validated.
+                     If not provided, check within PATH.
+        :rtype: None if check succeds, otherwise the error message containing the missing executable name.
+        """
+        if not self.contains_executable:
+            return
+        for program in self.contains_executable:
+            if not shutil.which(program, path=path):
+                return f"does not contain '{program}'"
```

Comment:
```suggestion
    def _check_executables(self, path: OptionalType[str] = None) -> Iterator[Result]:
        """Check that all executables from contains_executable list exist within the given directory or within PATH.

        :param path: optional path to the base directory against which executables will be validated.
                     If not provided, check within PATH.
        :rtype: None if check succeds, otherwise the error message containing the missing executable name.
        """
        for program in self.contains_executable or []:
            if not shutil.which(program, path=path):
                yield Result(False, f"does not contain '{program}'")
```

And on top of that update the docstring